HLTrigger/Configuration/python/customizer_AlpakaPixelTracking_Phase2_141X.py

```diff
@@ -0,0 +1,247 @@
+            appendToDataLabel = cms.string(''),
+            alpaka = cms.untracked.PSet(
+                backend = cms.untracked.string('')
+        )
+    )
+    process.hltESPSiPixelGainCalibrationForHLTSoA = cms.ESProducer('SiPixelGainCalibrationForHLTSoAESProducer@alpaka',
+            appendToDataLabel = cms.string(''),
+            alpaka = cms.untracked.PSet(
+                backend = cms.untracked.string('')
+        )
+    )
+    process.hltESPPixelCPEFastParamsPhase2 = cms.ESProducer('PixelCPEFastParamsESProducerAlpakaPhase2@alpaka', 
+            ComponentName = cms.string("PixelCPEFastParamsPhase2"),
+            appendToDataLabel = cms.string(''),
+            alpaka = cms.untracked.PSet(
+                backend = cms.untracked.string('')
+        )
+    )
+    process.hltOnlineBeamSpotDevice = cms.EDProducer('BeamSpotDeviceProducer@alpaka',
+            src = cms.InputTag('hltOnlineBeamSpot'),
+            alpaka = cms.untracked.PSet(
+                backend = cms.untracked.string('')
+        )
+    )
+    process.siPixelClustersSoA = cms.EDProducer('SiPixelPhase2DigiToCluster@alpaka',
+        mightGet = cms.optional.untracked.vstring,
+        alpaka = cms.untracked.PSet(
+            backend = cms.untracked.string('')
+        )
+    )
+    process.siPixelClusters = cms.EDProducer('SiPixelDigisClustersFromSoAAlpakaPhase2',
+        src = cms.InputTag('siPixelClustersSoA'),
+        clusterThreshold_layer1 = cms.int32(4000),
+        clusterThreshold_otherLayers = cms.int32(4000),
+        produceDigis = cms.bool(False),
+        storeDigis = cms.bool(False)
+    )
+    process.siPixelClusterShapeCache = cms.EDProducer('SiPixelClusterShapeCacheProducer',
+        src = cms.InputTag('siPixelClusters' ),
+        onDemand = cms.bool( False )
+    )
+    process.siPixelRecHitsSoA = cms.EDProducer('SiPixelRecHitAlpakaPhase2@alpaka',
+        beamSpot = cms.InputTag('hltOnlineBeamSpotDevice'),
+        src = cms.InputTag('siPixelClustersSoA'),
+        CPE = cms.string('PixelCPEFastParamsPhase2'),
+        mightGet = cms.optional.untracked.vstring,
+        # autoselect the alpaka backend
+        alpaka = cms.untracked.PSet(
+            backend = cms.untracked.string('')
+        )
+    )
+    process.siPixelRecHits = cms.EDProducer('SiPixelRecHitFromSoAAlpakaPhase2',
+        pixelRecHitSrc = cms.InputTag('siPixelRecHitsSoA'),
+        src = cms.InputTag('siPixelClusters'),
+    )
+
+    process.HLTDoLocalPixelSequence = cms.Sequence(
+         process.siPixelClustersSoA
+        +process.siPixelClusters
+        +process.siPixelClusterShapeCache  # should we disable this? Still needed by tracker muons
+        #+process.siPixelDigis             # not needed when copying digis from sim
+        +process.siPixelRecHitsSoA
+        +process.siPixelRecHits
+    )
+    process.HLTDoLocalStripSequence = cms.Sequence(
+        process.siPhase2Clusters
+    )
+    process.itLocalRecoSequence = cms.Sequence(
```

Comment:
Seuqences run at HLT start their label with ```HLT``` - please do so also for Phase-2.

Reply:
There are several modules and sequences in the phase2 menu which do not follow the naming convention. This problem is known, it will be fixed in a future PR. This sequence in particular is already defined in https://github.com/cms-sw/cmssw/blob/0cbc4b00b1664496105480e9f561a29ba7c9a68b/HLTrigger/Configuration/python/HLT_75e33/sequences/HLTDoLocalPixelSequence_cfi.py#L4 and I removed it from the customiser, since the default definition is ok (208d97d6f5c4dc677b0ce3d11e5d5db357e97785).

Reply:
Mmmh, as long as the new instances are used only in the customization function why do you need to keep the naming used elsewhere? Aren't you replacing the sequence entirely?

Reply:
the idea was to allow other POGs to use the customiser with minimal effort. Some paths include this sequence (for example https://github.com/cms-sw/cmssw/blob/0cbc4b00b1664496105480e9f561a29ba7c9a68b/HLTrigger/Configuration/python/HLT_75e33/paths/HLT_Mu37_Mu27_FromL1TkMuon_cfi.py#L19, but it is true for other sequences as well), and using a new name for the sequence would not propagate the changes done in the customiser to these paths